### PR TITLE
embassy-stm32: added readme for crates release

### PR DESCRIPTION
**Pull request by barnabywalters**

### CODE REVIEW

File: embassy-stm32/README.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+In practice, this works as follows:
+
+1. You tell the compiler which chip you’re using with a feature flag
+1. The stm32-metapac module generates register types for that chip at compile time, based on data from the stm32-data module
+1. The embassy-stm32 HAL picks the correct implementation each peripheral based on automatically-generated feature flags, and applies any other tweaks which are required for the HAL to work on that chip
+
+Be aware that, while embassy-stm32 strives to consistently support all peripherals across all chips, this approach can lead to slightly different APIs and capabilities being available on different families. Check the [documentation](https://docs.embassy.dev/embassy-stm32/) for the specific chip you’re using to confirm exactly what’s available.
+
+## Minimum supported Rust version (MSRV)
```
Comment (Dirbaio) [inline embassy-stm32/README.md:22]: in #2435 I decided not to duplicate the MSRV in all crate readmes (the policy is the same for all embassy crates and is documented in the repo readme). could you remove it?

File: embassy-stm32/README.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+Embassy is guaranteed to compile on the latest stable Rust version at the time of release. It might compile with older versions but that may change in any new patch release.
+
+## embassy-time Timer Driver
```
Comment (Dirbaio) [inline embassy-stm32/README.md:25]: timer driver -> time driver

Comment (barnabywalters) [reply embassy-stm32/README.md:25]: :+1:  guess that needs correcting in the docs too then https://embassy.dev/book/dev/stm32.html#_timer_driver

File: embassy-stm32/README.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+In practice, this works as follows:
+
+1. You tell the compiler which chip you’re using with a feature flag
+1. The stm32-metapac module generates register types for that chip at compile time, based on data from the stm32-data module
+1. The embassy-stm32 HAL picks the correct implementation each peripheral based on automatically-generated feature flags, and applies any other tweaks which are required for the HAL to work on that chip
+
+Be aware that, while embassy-stm32 strives to consistently support all peripherals across all chips, this approach can lead to slightly different APIs and capabilities being available on different families. Check the [documentation](https://docs.embassy.dev/embassy-stm32/) for the specific chip you’re using to confirm exactly what’s available.
+
+## embassy-time Time Driver
+If the `time` feature is enabled, embassy-stm32 provides a timer driver for use with [embassy-time](https://docs.embassy.dev/embassy-time/). You can pick which hardware timer is used for this internally via the `time-driver-*` features, or let embassy pick with `time-driver-any`.
```
Comment (Dirbaio) [inline embassy-stm32/README.md:23]: here too

Comment (Dirbaio) [inline embassy-stm32/README.md:23]: here too

**Pull request merged**